streamer/src/nonblocking/quic.rs

```diff
@@ -331,6 +331,7 @@ async fn run_server(
                 stats
                     .connection_rate_limited_across_all
                     .fetch_add(1, Ordering::Relaxed);
+                incoming.ignore();
```

Comment:
Yes, good point. I was consciously using ignore feature for two reasons: 1. it is less costly for the server to do so -- we do not have to schedule outgoing reply packets when it is already under load -- we have in the past excessive outgoing packets for under initial packets attacks. 2; the timeout will hopefully make the client taking longer time to do the retry -- it need to wait for the timeout before another attack. I think a non-collaborative client could do that regardless we do refuse or silently drop it. Maybe we could improve the logic in follow on PR to initially do the refuse and when it keep violating the limit we could just ignore.

Reply:
I think this is an issue because it makes well behaved clients retry, as to them this is effectively packet loss. The default packet loss timeout in quinn iirc is 1.3 * RTT. But I'm ok fixing this in a follow up!